packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -19,98 +20,69 @@ export interface LimitOwnProps {
     isLimitEditable?: boolean;
     isLimitTheGoalToReach?: boolean;
     isHistoryIncluded?: boolean;
+    onClick?: () => void;
 }
 
-const mapStateToProps = (state: IReactVaporState, {id, limit}: LimitOwnProps) => {
-    const ownLimit: number = limit ?? 100;
-    const currentLimit: number = +InputSelectors.getValue(state, {id}) || ownLimit;
-    return {
-        currentLimit,
-    };
-};
-
-const mapDispatchToProps = (dispatch: IDispatch, {id}: LimitOwnProps) => ({
-    onChangeLimit: (limitValue: string) => {
-        dispatch(changeInputValue(id, limitValue, true));
-    },
-});
+export interface LimitProps extends LimitOwnProps {}
 
-export interface LimitProps
-    extends LimitOwnProps,
-        ReturnType<typeof mapDispatchToProps>,
-        ReturnType<typeof mapStateToProps> {}
-
-type ContentProps = Pick<LimitProps, 'id' | 'usage' | 'limit' | 'isLimitEditable' | 'limitLabel' | 'onChangeLimit'>;
+type ContentProps = Pick<LimitProps, 'id' | 'usage' | 'limit' | 'isLimitEditable' | 'limitLabel'>;
 type UsageContentProps = Pick<ContentProps, 'usage'>;
-type LimitContentProps = Pick<
-    ContentProps,
-    'id' | 'limit' | 'usage' | 'isLimitEditable' | 'limitLabel' | 'onChangeLimit'
->;
-type HeaderProps = Pick<LimitProps, 'limitLabel' | 'isHistoryIncluded'>;
-type HistoryIconProps = Pick<HeaderProps, 'isHistoryIncluded'>;
+type LimitContentProps = Pick<ContentProps, 'id' | 'limit' | 'usage' | 'isLimitEditable' | 'limitLabel'>;
+type HeaderProps = Pick<LimitProps, 'limitLabel' | 'isHistoryIncluded' | 'onClick'>;
+type HistoryIconProps = Pick<HeaderProps, 'isHistoryIncluded' | 'onClick'>;
 type ProgressBarProps = Pick<LimitProps, 'usage' | 'isLimitTheGoalToReach' | 'limit'>;
```

Comment:
What do you think of removing all those types and simply sending all the props to all children?

E.g.,
```typescript
export const Limit: React.FunctionComponent<LimitProps> = (props) => {
    const {id, limit, className} = props;
    const {currentLimit} = useSelector((state: IReactVaporState) => ({
        currentLimit: +InputSelectors.getValue(state, {id}) || (limit ?? 100),
    }));

    return (
        <div className={classNames('limit-box mb2', className)}>
            <div className="limit-box-main p2 pb1">
                <HeaderDivision {...props} />
                <ContentDivision {...props} />
            </div>
            <ProgressBar {...props} />
        </div>
    );
};
```

Then all the children can use `LimitProps`

Reply:
Yeah I agree, I wasn't sure about all those types when I started refactoring, but it would be much cleaner